x-pack/packages/kbn-elastic-assistant/impl/assistant/conversations/conversation_settings/conversation_settings_editor.tsx

```diff
@@ -0,0 +1,328 @@
+ * 2.0.
+ */
+
+import { EuiFormRow, EuiLink } from '@elastic/eui';
+import React, { useCallback, useMemo } from 'react';
+
+import { HttpSetup } from '@kbn/core-http-browser';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { OpenAiProviderType } from '@kbn/stack-connectors-plugin/public/common';
+import { noop } from 'lodash/fp';
+import { Conversation, Prompt } from '../../../..';
+import * as i18n from './translations';
+import * as i18nModel from '../../../connectorland/models/model_selector/translations';
+
+import { ConnectorSelector } from '../../../connectorland/connector_selector';
+import { SelectSystemPrompt } from '../../prompt_editor/system_prompt/select_system_prompt';
+import { ModelSelector } from '../../../connectorland/models/model_selector/model_selector';
+import { useLoadConnectors } from '../../../connectorland/use_load_connectors';
+import { getGenAiConfig } from '../../../connectorland/helpers';
+import { ConversationsBulkActions } from '../../api';
+import { getDefaultSystemPrompt } from '../../use_conversation/helpers';
+
+export interface ConversationSettingsEditorProps {
+  allSystemPrompts: Prompt[];
+  conversationSettings: Record<string, Conversation>;
+  conversationsSettingsBulkActions: ConversationsBulkActions;
+  http: HttpSetup;
+  isDisabled?: boolean;
+  isFlyoutMode: boolean;
+  selectedConversation?: Conversation;
+  setConversationSettings: React.Dispatch<React.SetStateAction<Record<string, Conversation>>>;
+  setConversationsSettingsBulkActions: React.Dispatch<
+    React.SetStateAction<ConversationsBulkActions>
+  >;
+}
+
+/**
+ * Settings for adding/removing conversation and configuring default system prompt and connector.
+ */
+export const ConversationSettingsEditor: React.FC<ConversationSettingsEditorProps> = React.memo(
+  ({
+    allSystemPrompts,
+    selectedConversation,
+    conversationSettings,
+    http,
+    isDisabled = false,
+    isFlyoutMode,
+    setConversationSettings,
+    conversationsSettingsBulkActions,
+    setConversationsSettingsBulkActions,
+  }) => {
+    const { data: connectors, isSuccess: areConnectorsFetched } = useLoadConnectors({
+      http,
+    });
+
+    const selectedSystemPrompt = useMemo(() => {
+      return getDefaultSystemPrompt({ allSystemPrompts, conversation: selectedConversation });
+    }, [allSystemPrompts, selectedConversation]);
+    const handleOnSystemPromptSelectionChange = useCallback(
+      (systemPromptId?: string | undefined) => {
+        if (selectedConversation != null && selectedConversation.apiConfig) {
+          const updatedConversation = {
+            ...selectedConversation,
+            apiConfig: {
+              ...selectedConversation.apiConfig,
+              defaultSystemPromptId: systemPromptId,
+            },
+          };
+          setConversationSettings({
+            ...conversationSettings,
+            [updatedConversation.id || updatedConversation.title]: updatedConversation,
```

Comment:
Sometimes id can be an empty string:
<img width="2556" alt="additional conversation" src="https://github.com/elastic/kibana/assets/6295984/6e86c6e6-0d2d-473e-98d2-86a74b5f7e7f">